mutation/mutation_cleaner.hh

```diff
@@ -25,6 +25,9 @@ class mutation_cleaner_impl final {
         logalloc::allocating_section alloc_section;
         bool done = false; // true means the worker was abandoned and cannot access the mutation_cleaner_impl instance.
     };
+public:
+    // Callback arguments: change in total memory, change in used memory.
+    using on_space_freed_callback = std::function<void(size_t, size_t)>;
```

Comment:
Tracking total memory is dangerous, since it can change by memory compaction. I don't know if we account for that.